app/models/miq_set.rb

```diff
@@ -0,0 +1,66 @@
+class MiqSet < ApplicationRecord
```

Comment:
conceptually, I think of miq set as a collection of objects - e.g.: policies
I don't think of it as having its own table with a name and description. VmSet has a name and description of 'set of vms'?

avoiding a habtm and using a has many through is good
But I was expecting membership to just have 2 belongs to references - one being polymorphic if necessary.

Kinda feels like we are reinventing a rails association here.

Reply:
This is just STI for the left hand side of the association.  I guess we could get rid of it, but then you'd still need to know how to query specific types of sets.  Either that or we creates a "set" shaped table for every left hand side.  

> But I was expecting membership to just have 2 belongs to references - one being polymorphic if necessary.

That's exactly what MiqSetMembership is.